src/deck/drivers/interface/aideck.h

```diff
@@ -27,33 +27,74 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  *
- * aideck.h: AI-deck interface
+ * aideck.h: AI-deck/CPX interface
  */
 
 #pragma once
 #include <stdint.h>
 #include <stdbool.h>
 
-#include "routing_info.h"
-
 #define AIDECK_UART_TRANSPORT_MTU 100
 
+typedef enum {
+  STM32 = 1,
+  ESP32 = 2,
+  HOST = 3,
+  GAP8 = 4
+} __attribute__((packed)) CPXTarget_t;
+
+typedef enum {
+  SYSTEM = 1,
+  CONSOLE = 2,
+  CRTP = 3,
+  WIFI_CTRL = 4,
+  APP = 5,
+  TEST = 0x0E,
+  BOOTLOADER = 0x0F,
+} __attribute__((packed)) CPXFunction_t;
+
 typedef struct {
-    uint8_t start; // Should be 0xFF
-    uint8_t length; // Length of data
-    uint8_t data[AIDECK_UART_TRANSPORT_MTU];
-} __attribute__((packed)) uart_transport_packet_t;
+  CPXTarget_t destination : 4;
```

Comment:
So max 7 targets?

Reply:
It should be 16

Reply:
yes, that is right, 2^4, my mistake. Maybe we should write a comment about this next to the enum defining targets? To make it more clear?

Reply:
Yeah, this is a bit sneaky. The structs were initially done when there was not step in between the user assembling the data and sending it, i.e the struct was the same as on the wire. But now since we're copying it into the wire structure we could skip the enums being packed (and the full struct being packed) and fix this in the cpx send/receive. With asserts making sure that the target/function abides by the correct sizes. Yeah, we should probably do that.